examples/src/bin/rpk_example.rs

```diff
@@ -0,0 +1,390 @@
+        trace!(
+            "Supported schemes: {:?}",
+            self.supported_algs.supported_schemes()
+        );
+        self.supported_algs.supported_schemes()
+    }
+
+    fn requires_raw_public_keys(&self) -> bool {
+        true
+    }
+}
+
+#[derive(Debug)]
+struct SimpleRpkServerCertVerifier {
+    trusted_spki: Vec<SubjectPublicKeyInfoDer<'static>>,
+    supported_algs: WebPkiSupportedAlgorithms,
+}
+
+impl SimpleRpkServerCertVerifier {
+    fn new(trusted_spki: Vec<SubjectPublicKeyInfoDer<'static>>) -> Self {
+        let provider = Arc::new(provider::default_provider());
+        SimpleRpkServerCertVerifier {
+            trusted_spki,
+            supported_algs: provider
+                .clone()
+                .signature_verification_algorithms,
+        }
+    }
+}
+
+impl ServerCertVerifier for SimpleRpkServerCertVerifier {
+    fn verify_server_cert(
+        &self,
+        end_entity: &CertificateDer<'_>,
+        _intermediates: &[CertificateDer<'_>],
+        _server_name: &ServerName<'_>,
+        _ocsp_response: &[u8],
+        _now: UnixTime,
+    ) -> Result<ServerCertVerified, rustls::Error> {
+        let end_entity_as_spki = SubjectPublicKeyInfoDer::from(end_entity.as_ref());
+        match self
+            .trusted_spki
+            .contains(&end_entity_as_spki)
+        {
+            false => {
+                trace!("Server certificate not trusted");
+                Err(rustls::Error::NoCertificatesPresented)
+            }
+            true => {
+                trace!("Server certificate trusted");
+                Ok(ServerCertVerified::assertion())
+            }
+        }
+    }
+
+    fn verify_tls12_signature(
+        &self,
+        message: &[u8],
+        cert: &CertificateDer<'_>,
+        dss: &DigitallySignedStruct,
+    ) -> Result<HandshakeSignatureValid, rustls::Error> {
+        trace!("Verifying TLS 1.2 client signature");
+        verify_tls12_signature_with_spki(
+            message,
+            &SubjectPublicKeyInfoDer::from(cert.as_ref()),
+            dss,
+            &self.supported_algs,
+        )
+    }
+
+    fn verify_tls13_signature(
+        &self,
+        message: &[u8],
+        cert: &CertificateDer<'_>,
+        dss: &DigitallySignedStruct,
+    ) -> Result<HandshakeSignatureValid, rustls::Error> {
+        trace!("Verifying TLS 1.3 client signature");
+        verify_tls13_signature_with_spki(
+            message,
+            &SubjectPublicKeyInfoDer::from(cert.as_ref()),
+            dss,
+            &self.supported_algs,
+        )
+    }
+
+    fn supported_verify_schemes(&self) -> Vec<SignatureScheme> {
+        trace!(
+            "Supported schemes: {:?}",
+            self.supported_algs.supported_schemes()
+        );
+        self.supported_algs.supported_schemes()
+    }
+
+    fn requires_raw_public_keys(&self) -> bool {
+        true
+    }
+}
+
+fn main() -> Result<(), io::Error> {
```

Comment:
I think this needs to be rearranged into a "top down" style where `main` is up top and all of the supporting functions/types are below it.